src/main/java/com/microfocus/application/automation/tools/srf/run/RunFromSrfBuilder.java

```diff
@@ -480,19 +480,43 @@ private JSONArray getJobIds(JSONArray jobs) {
         int cnt = jobs.size();
         for (int k = 0; k < cnt; k++ ){
             JSONObject job = jobs.getJSONObject(k);
-            JSONObject jobExecutionError = job.getJSONObject("error");
-            if (jobExecutionError.size() != 0) {
-                JSONObject errorParameters = jobExecutionError.getJSONObject("parameters");
-                jobIds.add(errorParameters.getString("jobId"));
-                runningCount.add(errorParameters.getString("testRunId"));
-            } else {
-                jobIds.add(job.getString("jobId"));
-                runningCount.add(job.getString("testRunId"));
+            try {
+                if (job.has("error")) {
+                    String errorClassName = job.get("error").getClass().getSimpleName();
+                    switch (errorClassName) {
+                        case "JSONObject":
+                            JSONObject jobExecutionError = job.getJSONObject("error");
+                            handleJobError(jobIds, jobExecutionError);
+                            break;
+                        case "JSONArray":
+                            JSONArray jobExecutionErrors = job.getJSONArray("error");
+                            for (Object jobError : jobExecutionErrors) {
+                                JSONObject error = (JSONObject) jobError;
+                                handleJobError(jobIds, error);
+                            }
+                            break;
+                    }
+                } else {
+                    jobIds.add(job.getString("jobId"));
+                    runningCount.add(job.getString("testRunId"));
+                }
+            } catch (Exception e) {
```

Comment:
Don't you think we shouldn't catch all types of exceptions, only those that are related to your code?